documentation/src/docs/recovery-procedures/table-segment-recover-pre-index-full.md

```diff
@@ -0,0 +1,97 @@
+as soon as data for that Table Segment is processed and moved to LTS. For more details, please see 
+[PDP 48](https://github.com/pravega/pravega/wiki/PDP-48-(Key-Value-Tables-Beta-2)).
+
+The offline approach to build the Attribute Index adopted in Hash-based Table Segments leads to having a backlog of
+PUT and/or REMOVE operations that are stored in Tier-1, but not yet indexed and stored in LTS. If LTS is unavailable for
+a long period of time and one of such Hash-based Table Segments continues to receive load, there is a risk of having too
+much data being accumulated in Tier-1 only. This may be specially problematic as keys and values stored in Tier-1 are also
+cached in memory to guarantee consistency, in a process called Table Segment "pre-index". Due to this, there are few
+parameters that limit the amount of non-indexed (i.e., written in Tier-1 but not yet to LTS) data that Hash-based Table 
+Segments can tolerate: 
+- `tables.preindex.bytes.max` (_defaults to 128MB_): The maximum un-indexed data length for which a Hash-based Table Segment can 
+perform tail-caching.
+- `tables.recovery.timeout.millis` (_defaults to 60 seconds_): The maximum amount of time to wait for a Table Segment Recovery.
+- `tables.unindexed.bytes.max` (_defaults to 128MB_): The maximum allowed un-indexed data length for a Segment. Any updates that 
+occur while the amount of un-indexed data exceeds this threshold will be blocked until the amount of un-indexed data is reduced. 
+- `tables.systemcritical.unindexed.bytes.max` (_defaults to 256MB_): Same as the previous parameter, but for Metadata
+Table Segments (i.e., `container_metadata` and  `storage_metadata` on each Segment Container).
+
+In rare cases in which LTS is super slow or unavailable for hours, but there is some activity happening in a Hash-based
+Table Segment, we can find errors in the logs such as:
+
+```
+2022-08-08 13:07:57,194 1192230051 [core-19] INFO i.p.s.s.tables.ContainerKeyIndex - KeyIndex[27]: Table Segment 14 cannot perform tail-caching because tail index too long (150542984).
+```
+```
+2022-08-08 13:07:57,194 1192230051 [core-19] INFO i.p.s.s.tables.ContainerKeyIndex - KeyIndex[27]:: System-critical TableSegment 3 is blocked due to reaching max unindexed size.
+```
+
+Messages like the ones above indicate that Table Segments may not be able to recover, to write new entries, or both.
+If the impacted Hash-based Table Segments are critical for the operation of Pravega, it may leave the impacted Segment
+Container unable for processing new operations. Once reached this situation, recovering the Segment Container requires 
+manual intervention.
+
+# Repair Procedure
+
+The repair procedure is relatively simple and works as follows:
+
+1. _Make sure that LTS is available and works properly_: The root cause of the problem is related to LTS not being able
+to store data fast enough, of just being unavailable for a very long period of time. Any attempt to recover Pravega
```

Comment:
to store data fast enough, [of ] just being unavailable  -- > you mean "or just being unavaible...."